<a href="https://colab.research.google.com/github/AIexndr/MTaN/blob/main/GB_DZ_2_Matan.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Даны 4 последовательности.



Необходимо: исследовать их на монотонность; исследовать на ограниченность; найти пятый по счету член.





In [2]:
import math
n = 5 # пятый член последовательности вычисляем подстановкой в формулу
print(2 ** n - n)
print( 1 / (1 - (n + 1)))
print( -1 ** n + math.sqrt(2 * n))
print( (-1) ** (2 * n) + 1 / (n**2))

27
-0.2
2.1622776601683795
1.04


${{a_n}}^\infty_{n=1} = 2^n - n$ - возрастающая, не ограничена  $a_5 = 27$</p>

${{b_n}}^\infty_{n=2} = \frac{1}{1-n}$ - возрастающая, ограниченная  $b_5 = -0.2$

${{c_n}}^\infty_{n=1} = -1^n + \sqrt{2n}$ -  возрастает, не ограниченна $c_5 = 2.162$

${{d_n}}^\infty_{n=1} = (-1)^{2n} + \frac{1}{n^2}$ -  убывает,  ограниченна   $d_5 = 1.04$

Найти 12-й член заданной неявно последовательности

$a_1 = 128, a_{n+1} - a_n = 6$ $a_{12} = a_1 + (12-1)(a_{n+1} - a_n) = 194$

$a_1 = 128, a_{n+1} - a_n = 6$ $  a_{12} = a_1 + (12-1)(a_{n+1} - a_n) = 194$

In [3]:
128+11*6

194

На языке Python предложить алгоритм вычисляющий численно предел
  $\lim _{n \rightarrow+\infty} \frac{n}{\sqrt[n]{n !}}$с точностью $\varepsilon=10^{-7}$

формула Стирлинга (также формула Муавра — Стирлинга) для аппроксимации функции факториала
$$n ! \approx \sqrt{2 \pi n}\left(\frac{n}{e}\right)^{n}$$
После подстановки формулы Стирлинга в функцию получаем:
$$\lim _{n \rightarrow+\infty} \frac{e}{\sqrt[2n]{\pi n}}$$

In [8]:
import numpy as np
import itertools
def f(n):
    return (np.exp(1))/((np.pi*n)**(1/(2*n)))

Так как значения функции Стирлинга имеют приближенное значение к значению функции факториала, будем искать такие два ближайшие значения функции, разность которых меньше 1e-09

In [9]:
prev = f(1)
for i in range(2, 200000000):
    new_val = f(i)
    if new_val - prev >= 1e-09:
        prev = new_val
    else:
        print(i, new_val, prev)
        break

127163 2.7181439763031636 2.7181439753031715
